File: 04_Managing_Indexes/02_queryGSI.js
var params = {
    TableName: 'productTableGSI',
    IndexName: "NameManfrIndex",
    KeyConditionExpression: '#nm = :value', // a string representing a constraint on the attribute
    ExpressionAttributeNames: { // a map of substitutions for attribute names with special characters
        '#nm': 'name'
    },
    ExpressionAttributeValues: { // a map of substitutions for all attribute values
      ':value': 'S3'
    }
};
docClient.query(params, function(err, data) {
    if (err) ppJson(err); // an error occurred
    else ppJson(data); // successful response

});

File: 04_Managing_Indexes/02_QueryGSI.cs
client = new AmazonDynamoDBClient();

QueryRequest queryRequest = new QueryRequest
{
    TableName = "productTableGSI",
    IndexName = "NameManfrIndex",
    KeyConditionExpression = "#nm = :v_name",
    ExpressionAttributeNames = new Dictionary<String, String> {
        {"#nm", "name"}
    },
    ExpressionAttributeValues = new Dictionary< string, AttributeValue> {
        {":v_name", new AttributeValue { S =  "S3" }}
    }
    };

    var result = client.Query(queryRequest);